### ISSUE TITLE
How about a message type which will never trigger notifications?

### ISSUE THREAD

**Issue by rowskilled14** [matrix-org/matrix-spec#2031]
**Suggestion**
Something like `/silent` followed by a message, being the client usage.  It of course will need a Matrix spec to support the functionality before clients can implement that type of format.

The idea being, `/silent  test message 123` would never trigger a notification for the other user, pending their specific settings.  Keyword notifications or perhaps a special setting on their end could force override this rule.  

___

The use case for this could extend to many scenarios, but would typically revolve around the idea of: "I have something I want to tell the other user, but it isn't important right now and I have a particular reason to not bother them at this very moment, yet I still want to send the message right now."  It could be that you know someone is in a meeting, someone is at an appointment, or it's very late at night and they're probably asleep.  In each case, you might want the message to be ready and waiting for the user as soon as they're free, but you don't want to be a nuisance in the meantime.  

Some bot/scripting/automated/API functionality could be utilized from this as well for similar lines of reasoning.  

The message would show up with slightly different formatting in clients to indicate that it was a silent message.  Something resembling a whisper type message from other chat systems.  Nothing massively distracting, but just a small indicator of sorts.

Users could specify in their client a specific setting for how to receive these messages as touched on above.  These settings could include: 

* The ability to fully disable this feature on a per-room basis, a per-user basis, and globally
* The ability to choose between:
  *  silent notifications still visually appear, but are silent/not noisy
  * OR
  *  silent notifications not appearing at all until the user manually checks the client
* The ability to listen for keywords to override this feature

I'd be happy to submit an MSC if I could receive some guidance on how to approach this.  I'm sure I'm overlooking a lot of definitions and potential issues, and I'm open to real feedback and discussion.  

Thank you very much!

**Comment by richvdh** [comment]
This is [`m.notice`](https://spec.matrix.org/v1.13/client-server-api/#mnotice)

**Comment by rowskilled14** [comment]
Is that really the same?  m.notice is intended for automated clients and not a real user's real client.  As such, no clients have implemented it.  Even further, it only supports text input and no other forms of messages are allowed.  m.notice is *always* formatted differently as well, and doesn't respect the outlined potential settings as mentioned above, but is always formatted such that text is "from a bot." 

I think it's quite distinct.  The fact that m.notice isn't intended for real users to use alone is fully distinct in my opinion.  The use cases are entirely separate with zero overlap.

**Comment by richvdh** [comment]
Fair enough. I'm yet to be convinced that there is enough of a usecase for another message type, but I'll reopen this.

> I also don't see anything in the spec specifying that this should be treated as a silent notification - in fact, it explicitly says the opposite (as it explicitly says, `An m.notice message must be treated the same way as a regular m.text message with two exceptions.` and neither exception is about notifications.).

The difference in notifications comes about by virtue of the definition of the [default push rules](https://spec.matrix.org/v1.13/client-server-api/#default-override-rules), which include a rule that specifically excludes `m.notice` from notifications. Arguably that's not an *exception* specific to `m.notice`: it's just the result of applying the default push rules to the event as you would with any other event. Certainly the spec could be clearer here.

**Comment by rowskilled14** [comment]
Thank you for the clarification.

___

Perhaps instead of an entirely separate message type, this could simply be a new content field for existing message types?  m.text, m.image, m.video, (and others where relevant) could have a new (optional) content name of, "silent" and of type, "boolean" or something similar.  

Alternatively, perhaps m.notice could be expanded to support real user's input in real clients, and support other data types besides simple text.  Currently the scope of the intent seems to imply this won't happen (as of now).


___

The purpose of expanding beyond just text data fields would be to better support automated messages as well.  In automated situations where it is desirable to send text, images, and video alongside each other, currently three messages are needed and thus will trigger three (or two if you use m.notice for the text) notifications.  A way to silent two of the three messages could change that, and would surely find other beneficial use cases as well.


___

I think any of these three approaches would more or less satisfy the core of what I'm seeking.  I really think this would be a useful feature that plenty of others could see benefit from.

**Comment by KitsuneRal** [comment]
The case with three messages is addressed by extensible events. I agree that `m.notice` is not really expected to be sent by a human. An MSC for this is welcome.
